addons/website/static/src/js/content/menu.js

```diff
@@ -870,11 +870,38 @@ publicWidget.registry.HeaderGeneral = publicWidget.Widget.extend({
         "show.bs.offcanvas #top_menu_collapse, #top_menu_collapse_mobile": "_onCollapseShow",
         "hidden.bs.offcanvas #top_menu_collapse, #top_menu_collapse_mobile": "_onCollapseHidden",
     },
-
+    /**
+     * @override
+     */
+    start() {
+        this._updateActiveMenuLinks();
+        return this._super(...arguments);
+    },
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     */
+    _updateActiveMenuLinks() {
+        const currentPathName = window.location.pathname;
+        const topMenuEl = this.el.querySelector("#top_menu");
+        if (topMenuEl) {
+            // Check and update the active state of menu items based on the current page
+            const dropdownMenusEls = this.el.querySelectorAll(".o_mega_menu, .dropdown-menu");
+            dropdownMenusEls.forEach((dropdownMenusEl) => {
+                const linksEls = dropdownMenusEl.querySelectorAll(`a:not([href="#"])`);
+                linksEls.forEach((el) => {
+                    const href = new URL(el.href);
```

Comment:
Do you really need this? Isn’t `el.pathname` enough?

Reply:
`el.pathname`?? `el` is an instance of HTMLElement... not an URL instance.